src/client.rs

```diff
@@ -16,6 +16,7 @@ const CR: u8 = 0x0d;
 const LF: u8 = 0x0a;
 
 /// Stream to interface with the IMAP server. This interface is only for the command stream.
+#[derive(Debug)]
```

Comment:
This seems fine, though I'm unsure why having the stream implement `Debug` is useful?

Reply:
One other advantage of implementing traits eagerly, i.e. if there is no deliberate reason or cause not to, is that a compile time error will occur if the type is extended to include elements/fields that do not implement that common trait. Then the breaking change is pointed out and it can be reconsidered.

Reply:
Seems fine.